### PR TITLE
create EmailDirectAnswer interface

### PR DESCRIPTION
**Pull request by juliannzhou**
This PR creates EmailDirectAnswer interface for direct answers with email values and an email type to define generic type variable.

J=SLAP-2313
TEST=manual

Test: created a test interface with EmailDirectAnswer and works as expected for when inputting email type values. The email type is verified by checking api response in Search.

### CODE REVIEW

File: src/models/searchservice/response/EmailDirectAnswer.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+export interface EmailDirectAnswer extends FieldValueDirectAnswer<Email> {
+  fieldType: BuiltInFieldType.Email
+}
+
+/**
+ * A type to use in email direct answer's value {@link FieldValueDirectAnswer.value}.
+ *
+ * @public
+ */
+export type Email = string[];
```
Comment (oshi97) [inline src/models/searchservice/response/EmailDirectAnswer.ts:18]: I like the idea of making a type in general, but here it feels a little unnecessary? I would probably just do FieldValueDirectAnswer<string[]>

Comment (oshi97) [reply src/models/searchservice/response/EmailDirectAnswer.ts:18]: also is email always `string[]` or can it also just be `string`?

Comment (juliannzhou) [reply src/models/searchservice/response/EmailDirectAnswer.ts:18]: From what I've tested, they all returned `string[]` including results that only have one email.

File: src/models/searchservice/response/EmailDirectAnswer.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { BuiltInFieldType } from './directanswer/BuiltInFieldType';
+import { DirectAnswer } from './directanswer/DirectAnswer';
+
+/**
+ * A {@link DirectAnswer} interface with email value.
+ *
+ * @public
+ */
+export interface EmailDirectAnswer extends DirectAnswer<string[]> {
```
Comment (nmanu1) [inline src/models/searchservice/response/EmailDirectAnswer.ts:9]: should this extend `BaseFieldValueDirectAnswer`?

File: src/models/searchservice/response/EmailDirectAnswer.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { BuiltInFieldType } from './directanswer/BuiltInFieldType';
+import { BaseFieldValueDirectAnswer } from './directanswer/BaseFieldValueDirectAnswer';
+
+/**
+ * A {@link BaseFieldValueDirectAnswer} interface with email value.
+ *
+ * @public
+ */
+export interface EmailDirectAnswer extends BaseFieldValueDirectAnswer<string[]> {
```
Comment (yen-tt) [inline src/models/searchservice/response/EmailDirectAnswer.ts:9]: can we rename this to EmailFieldValueDirectAnswer?

File: src/models/searchservice/response/EmailFieldValueDirectAnswer.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import { BuiltInFieldType } from './directanswer/BuiltInFieldType';
+import { BaseFieldValueDirectAnswer } from './directanswer/BaseFieldValueDirectAnswer';
+
+/**
+ * A {@link BaseFieldValueDirectAnswer} interface with email value.
```
Comment (yen-tt) [inline src/models/searchservice/response/EmailFieldValueDirectAnswer.ts:5]: nit: can we say "A {@link BaseFieldValueDirectAnswer} with 'email' field type" instead of "...email value"?

File: src/models/questionsubmission/BuiltInFieldType.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+/**
+ * Possible built-in field types for {@link DirectAnswer.fieldType}.
+ * @public
+ */
+export enum BuiltInFieldType {
```
Comment (yen-tt) [inline src/models/questionsubmission/BuiltInFieldType.ts:5]: why is this added to questionsubmission?

Comment (juliannzhou) [reply src/models/questionsubmission/BuiltInFieldType.ts:5]: I think I accidentally dragged it in there for the last commit.

File: src/models/index.ts [hunk 6]
```diff
@@ -79,6 +79,7 @@ export { VerticalSearchResponse } from './searchservice/response/VerticalSearchR
 export { Snippet } from './searchservice/response/Snippet';
 export { ErrorType } from './searchservice/response/ErrorType';
 export { FailedVertical } from './searchservice/response/FailedVertical';
+export { EmailFieldValueDirectAnswer } from './searchservice/response/EmailFieldValueDirectAnswer';
```
Comment (yen-tt) [inline src/models/index.ts:82]: nit: can we update this branch with the changes in develop and add this to where the other FieldValue interfaces are in the file?

File: test-site/src/ts/initDirectAnswers.ts [hunk 7]
```diff
@@ -124,6 +124,9 @@ function ensureFieldValueSwitchability(da: FieldValueDirectAnswer) {
     case BuiltInFieldType.Integer:
       console.log(da.value);
       break;
+    case BuiltInFieldType.Email:
+      console.log(da.value);
```
Comment (yen-tt) [inline test-site/src/ts/initDirectAnswers.ts:128]: should add a query example above for email field

File: src/models/searchservice/response/directanswer/FieldValueDirectAnswer.ts [hunk 8]
```diff
@@ -30,4 +31,5 @@ export type FieldValueDirectAnswer =
   IosAppUrlDirectAnswer |
   AndroidAppUrlDirectAnswer |
   ComplexUrlDirectAnswer |
-  IntegerDirectAnswer;
+  IntegerDirectAnswer |
```
Comment (yen-tt) [inline src/models/searchservice/response/directanswer/FieldValueDirectAnswer.ts:34]: trailing spaces

**Review by yen-tt — APPROVED** [review]
